src/context/VendingMachineContext.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useState, useMemo } from "react";
+
+const VendingMachineContext = React.createContext();
+export const useVendingMachineContext = () => useContext(VendingMachineContext);
+
+function VendingMachineProvider({ children }) {
+    const [record, setRecord] = useState([]);
+    const [moneyInVendingMachine, setMoneyInVendingMachine] = useState({});
+
+    const addRecord = (newRecord) => {
+        setRecord([...record, newRecord]);
+    };
+
```

Comment:
실무에서는 이정도까지는 아니지만 학습하시는 중이시기 떄문에 한가지 공부해볼만한 키워드 드립니다!

현재 addRecord 는 record 라는 상태를 참조하고 있기떄문에, record 상태가 변경될때 마다 재정의 됩니다!
함수의 재정의(재렌더링)를 막기 위해 useCallback 을 사용할 수 있는데, 이러면 setRecord 가 참조하고 있는 record 가 변경되지 않기 때문에 로직에 문제가 생깁니다.

이떄 사용하는 것이 setState 의 함수형 업데이트 입니다. 해당 로직을 이렇게 변경할 수도 있겠네요!

```suggestion
const addRecord = useCallback((newRecord) => {
    setRecord(prevRecord => ([...prevRecord, newRecord]));
}, []);
```
위 같이 함수를 구성하게 되면, 더 이상 addRecord 함수가 record 라는 상태에 의존하지 않고, Provider 생성 단계에서 한번 만들어진 후 계속 재사용 될 수 있겠네요!